content/blog/2022-07-05-Practical-data-drift-monitoring-with-DVC.md

```diff
@@ -0,0 +1,337 @@
+transparent. What you can see in the figures below is how well the detectors
+learn the pattern in the data. The blue line shows the average relative ratio of
+rented bikes daily. The other lines show the reconstructed patterns from the
+detector. The more significant the difference between the input and output, the
+less accurate the detector is.
+
+![Effect of various latent_dim values on concept drift detector](/uploads/images/2022-07-05/training_concept_drift.png)
+
+# Detection of concept drift in data
+
+At this point, we have trained both the ML model and drift detectors and we are
+ready to deploy them. A detailed description of how to deploy these models is
+beyond scope of this blog post but you may find inspiration in the
+[MLEM blog post](https://dvc.org/blog/MLEM-release). To show how the concept
+drift detector performs on the ‘production’ data, we have prepared another DVC
+pipeline that is shown in the diagram below. This pipeline has only two stages.
+In the first stage, we prepare data that shall mimic production data. In the
+second stage, we generate plots. Note that the second stage depends on the
+concept drift detector generated by the training pipeline.
+
+```bash
+$ dvc dag concept_drift_demo/dvc.yaml
+           +------------------+
+           | data_preparation |
+           +------------------+
+                     *
+                     *
+                     *
+    +---------------------------------+
+    | concept_drift_detector_training |
+    +---------------------------------+
+                     *
+                     *
+                     *
++------------------------------------------------+
+| concept_drift_demo\dvc.yaml:evaluation_cd_demo |
++------------------------------------------------+
++------------------------------------------------------+
+| concept_drift_demo\dvc.yaml:data_preparation_cd_demo |
++------------------------------------------------------+
+```
+
+This pipeline shall represent the following imaginary business scenario. We
+train our ML model on January-April data, and it seems to work fine. Let’s
+modify the original dataset to simulate the following business scenario: _A new
+competitor in the area starts to offer significant discounts for afternoon hours
+during weekends. This will have a tremendous effect on the number of rented
+bikes. Estimate says that this decreased the number of rented bikes by 70%._ We
+can reproduce this scenario by running:
+
+```bash
+$ dvc repro concept_drift_demo/dvc.yaml
+```
+
+This is the use case where the concept drift detector shall detect that the
+pattern has changed. As shown in the figure below, the large reconstruction
+error of the detector causes the detector’s feature score to surpass the
+detector’s threshold. Therefore, we would detect that there is a concept drift.
+
+![Effect of various latent_dim values on concept drift detector](/uploads/images/2022-07-05/concept_drift_effect.png)
```

Comment:
@iterative/websites I believe image rendering is broken. It should be rendered as 1/2x, we had the logic before.

@Meleagos can you make all images 2x ? they look blurry at the moment on retina dispaly

Reply:
thanks @julieg18 ! can we merge it back here please?

Reply:
Merged!